src/components/ac-layer/ac-layer.component.ts

```diff
@@ -30,20 +30,25 @@ import { EllipsoidDrawerService } from '../../services/drawers/ellipoid-drawer/e
 import { PolylineVolumeDrawerService } from '../../services/drawers/polyline-volume-dawer/polyline-volume-drawer.service';
 import { WallDrawerService } from '../../services/drawers/wall-dawer/wall-drawer.service';
 import { RectangleDrawerService } from '../../services/drawers/rectangle-dawer/rectangle-drawer.service';
+import { MapLayersService } from '../../services/map-layers/map-layers.service';
```

Comment:
ac-layer should update map-layers service when it is destroyed. maps-layer.service should delete the relevant datasourcers